Packages/com.unity.inputsystem/InputSystem/Plugins/XR/TrackedPoseDriver.cs

```diff
@@ -616,6 +605,40 @@ bool HasStereoCamera(out Camera cameraComponent)
             return TryGetComponent(out cameraComponent) && cameraComponent.stereoEnabled;
         }
 
+        // Evaluates whether the given action has at least one resolved control and may generate input.
+        private static bool HasResolvedControl(InputAction action)
+        {
+            // Action cannot have controls if null.
+            if (action == null)
+                return false;
+
+            // Attempt to grab state and resolve bindings unless already resolved.
+            var actionMap = action.GetOrCreateActionMap();
+            actionMap.ResolveBindingsIfNecessary();
+            var state = actionMap.m_State;
+            if (state == null)
+                return false;
+
+            // Get list of resolved controls to determine if a device actually has a tracking state.
+            var actionIndex = action.m_ActionIndexInState;
+            var totalBindingCount = state.totalBindingCount;
+            for (var i = 0; i < totalBindingCount; ++i)
+            {
+                unsafe
+                {
+                    ref var bindingState = ref state.bindingStates[i];
+                    if (bindingState.actionIndex != actionIndex)
+                        continue;
+                    if (bindingState.isComposite)
+                        continue;
+                    if (bindingState.controlCount > 0)
+                        return true;
```

Comment:
While the [Unity-Meta project style guideline](https://github.cds.internal.unity3d.com/unity/unity-meta/blob/stable/ReferenceSource/CSharp/Assets/CSharpReference.cs#L231) doesn't explicitly call it out, putting spaces between un-braced conditional logic makes it more readable and less prone to accidental coding mistakes in the future.

```suggestion
                    if (bindingState.actionIndex != actionIndex)
                        continue;

                    if (bindingState.isComposite)
                        continue;

                    if (bindingState.controlCount > 0)
                        return true;
```

Reply:
I am not going to argue against that so will update accordingly. :) Will post with commit ID here once done.

Reply:
Addressed in https://github.com/Unity-Technologies/InputSystem/pull/1823/commits/6b5c6a2af9288169032ebf9b3364e8fd2de18ae7, feel free to resolve if sufficient.

Reply:
Realise now you did it as a suggestion, but commit should be equivalent.